### DISCUSSION TITLE
Seeking help to build TS libraries

### DISCUSSION THREAD

**Discussion by officer-kd6-3dot7** [trpc/trpc#6287] [General]
For a while now, I've been wanting to build tools to publish as libraries. While there’s an abundance of tutorials, guides, and blogs on how to build apps or projects, I’ve struggled to find similar resources for libraries. Things like, getting started, API design, recommended patterns, rules to follow, and best practices for building good TS libraries. If anyone can point me to helpful resources, I’d greatly appreciate it.

**Comment by KATT** [comment]
https://github.com/JoshuaKGoldberg/create-typescript-app
